src/models/model.cpp

```diff
@@ -65,7 +65,18 @@ void State::Run(OrtSession& session, int new_batch_size) {
   }
```

Comment:
In session.Run(...) above on line 58, we don't do anything on termination?

The issue is that there's a potential error race. If the user calls 'SetTerminate()' and then a non terminate error was thrown from session.Run() the user will not know it was a non termination related error (unless they look at the string).

Can we detect termination from session.Run? If so, we should rename session_terminated to session_terminate_set and have a second variable for "session_terminated" for if we actually hit the termination case (or if we call ThrowErrorIfSessionTerminated, as that will catch the non session.Run terminate cases).

@baijumeswani can review my thinking also.